packages/catalog-model/src/entity/policies/FieldFormatEntityPolicy.ts

```diff
@@ -50,7 +50,43 @@ export class FieldFormatEntityPolicy implements EntityPolicy {
       }
 
       if (!isValid) {
-        throw new Error(`${field} "${value}" is not valid`);
+        let expectation;
+        switch (validator.name) {
```

Comment:
bit of a stretch perhaps but to make this typesafe, you could do 

```ts
        switch (
          validator.name as
            | keyof typeof KubernetesValidatorFunctions
            | keyof typeof CommonValidatorFunctions
        ) {
 ```

:)

Reply:
> Cool, just nits. I think these are due for a rewrite such that each validator throws its own error message instead, but that's for a later time.

Yes! That was my initial plan but since i'm still a greenhorn in the OSS community I didn't want to muck around too much :P played it safe!

Reply:
> bit of a stretch perhaps but to make this typesafe, you could do
> 
> ```ts
>         switch (
>           validator.name as
>             | keyof typeof KubernetesValidatorFunctions
>             | keyof typeof CommonValidatorFunctions
>         ) {
> ```
> 
> :)

good catch! I'll implement it. Also felt that my switch was a bit sloppy